lms/djangoapps/bulk_email/tasks.py

```diff
@@ -119,20 +122,21 @@ def _get_recipient_queryset(user_id, to_option, course_id, course_location):
                 courseenrollment__course_id=course_id,
                 courseenrollment__is_active=True
             )
-            # Now we do some queryset sidestepping to avoid doing a DISTINCT
-            # query across the course staff and the enrolled students, which
-            # forces the creation of a temporary table in the db.
-            unenrolled_staff_qset = recipient_qset.exclude(
+
+            # to avoid duplicates, we only want to email unenrolled course staff
+            # members here
+            unenrolled_staff_qset = staff_instructor_qset.exclude(
                 courseenrollment__course_id=course_id, courseenrollment__is_active=True
             )
-            # use read_replica if available:
-            unenrolled_staff_qset = use_read_replica_if_available(unenrolled_staff_qset)
 
-            unenrolled_staff_ids = [user.id for user in unenrolled_staff_qset]
-            recipient_qset = enrollment_qset | User.objects.filter(id__in=unenrolled_staff_ids)
+            # use read_replica if available:
+            recipient_qsets = [
+                use_read_replica_if_available(unenrolled_staff_qset),
+                use_read_replica_if_available(enrollment_qset),
+            ]
 
     # again, use read_replica if available to lighten the load for large queries
-    return use_read_replica_if_available(recipient_qset)
+    return recipient_qsets
```

Comment:
Nit: setting `recipient_qsets` before the large `if` clause, and then setting it again inside the `if` clause seems confusing to me. I'd probably return in the `if`, and then return in an `else` (to be explicit that they're exclusive cases).